src/main/java/com/patchworkmc/mapping/IntermediaryHolder.java

```diff
@@ -0,0 +1,71 @@
+package com.patchworkmc.mapping;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+/**
+ * A lazily loading representation of all Minecraft names and descriptors.
+ */
+public class IntermediaryHolder {
```

Comment:
`IntermediaryHolder` isn't the best name since this isn't necessarily specific to intermediary. Where this is used could be documented as well.